tests/test_cases/test_cocotb/test_handle.py

```diff
@@ -66,21 +67,193 @@ async def test_delayed_assignment_still_errors(dut):
+    await int_overflow_test(dut.stream_in_data_39bit, value)
+
+
+@cocotb.test()
+async def test_int_39bit_underflow(dut):
+    """Test 39-bit vector underflow."""
+    value = gen_int_unfl_value(len(dut.stream_in_data_39bit))
+    await int_overflow_test(dut.stream_in_data_39bit, value)
+
+
+@cocotb.test()
+async def test_int_64bit(dut):
+    """Test int access to 64-bit vector."""
+    values = gen_int_test_values(len(dut.stream_in_data_wide))
+    await int_values_test(dut.stream_in_data_wide, values)
+
+
+@cocotb.test()
+async def test_int_64bit_overflow(dut):
+    """Test 64-bit vector overflow."""
+    value = gen_int_ovfl_value(len(dut.stream_in_data_wide))
+    await int_overflow_test(dut.stream_in_data_wide, value)
+
+
+@cocotb.test()
+async def test_int_64bit_underflow(dut):
+    """Test 64-bit vector underflow."""
+    value = gen_int_unfl_value(len(dut.stream_in_data_wide))
+    await int_overflow_test(dut.stream_in_data_wide, value)
+
+
+@cocotb.test()
+async def test_int_128bit(dut):
+    """Test int access to 128-bit vector."""
+    values = gen_int_test_values(len(dut.stream_in_data_dqword))
+    await int_values_test(dut.stream_in_data_dqword, values)
+
+
+@cocotb.test()
+async def test_int_128bit_overflow(dut):
+    """Test 128-bit vector overflow."""
+    value = gen_int_ovfl_value(len(dut.stream_in_data_dqword))
+    await int_overflow_test(dut.stream_in_data_dqword, value)
+
+
+@cocotb.test()
+async def test_int_128bit_underflow(dut):
+    """Test 128-bit vector underflow."""
+    value = gen_int_unfl_value(len(dut.stream_in_data_dqword))
+    await int_overflow_test(dut.stream_in_data_dqword, value)
+
+
 @cocotb.test(expect_error=AttributeError if cocotb.SIM_NAME in ["Icarus Verilog"] else ())
 async def test_integer(dut):
-    """
-    Test access to integers
-    """
-    log = logging.getLogger("cocotb.test")
-    await Timer(10, "ns")
-    dut.stream_in_int = 4
-    await Timer(10, "ns")
-    await Timer(10, "ns")
-    got_in = int(dut.stream_out_int)
-    got_out = int(dut.stream_in_int)
-    log.info("dut.stream_out_int = %d" % got_out)
-    log.info("dut.stream_in_int = %d" % got_in)
-    assert got_in == got_out, "stream_in_int and stream_out_int should not match"
+    """Test access to integers."""
+    values = gen_int_test_values(32, limits=_Limits.SIGNED_NBIT)
+    await int_values_test(dut.stream_in_int, values, limits=_Limits.SIGNED_NBIT)
+
+
+@cocotb.test(expect_error=AttributeError if cocotb.SIM_NAME in ["Icarus Verilog"] else ())
+async def test_integer_overflow(dut):
+    """Test integer overflow."""
+    value = gen_int_ovfl_value(32, limits=_Limits.SIGNED_NBIT)
+    await int_overflow_test(dut.stream_in_int, value)
+
+
+@cocotb.test(expect_error=AttributeError if cocotb.SIM_NAME in ["Icarus Verilog"] else ())
+async def test_integer_underflow(dut):
+    """Test integer underflow."""
+    value = gen_int_unfl_value(32, limits=_Limits.SIGNED_NBIT)
+    await int_overflow_test(dut.stream_in_int, value)
```

Comment:
This could be a `TestFactory` as well, but I think it doesn't handle passing arguments to `cocotb.test`. So nvm, just complaining =)